SConstruct

```diff
@@ -670,6 +670,8 @@ for variant_path in variant_paths:
         with gem5_scons.Configure(env) as conf:
             conf.CheckCxxFlag('-Wno-c99-designator')
             conf.CheckCxxFlag('-Wno-defaulted-function-deleted')
+            # Silence warning for DPRINTF_UNCONDITIONAL
+            conf.CheckCxxFlag('-Wno-compound-token-split-by-macro')
```

Comment:
I would prefer make (minor) changes to `DPRINTF_UNCONDITIONAL` instead of quieting a warning. How hard do you think that would be? What's the warning?

Reply:
That would definitely be better, let me check:

```
src/base/trace.test.cc:645:5: error: '}' and ')' tokens terminating statement expression appear in different macro expansion contexts [-Werror,-Wcompound-token-split-by-macro]
    DPRINTF_UNCONDITIONAL(TraceTestDebugFlag, "Test message");
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/base/trace.hh:253:5: note: expanded from macro 'DPRINTF_UNCONDITIONAL'
    GEM5_DEPRECATED_MACRO_STMT(DPRINTF_UNCONDITIONAL,      \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/base/compiler.hh:93:53: note: expanded from macro 'GEM5_DEPRECATED_MACRO_STMT'
     do {{definition;} GEM5_DEPRECATED_MACRO(name, {}, message);} while (0)
                                                    ^
src/base/compiler.hh:89:66: note: expanded from macro 'GEM5_DEPRECATED_MACRO'
     ([](){[[deprecated(message)]] int name{}; return name;}(), (definition))
                                                                 ^~~~~~~~~~
src/base/trace.test.cc:645:5: note: ')' token is here
    DPRINTF_UNCONDITIONAL(TraceTestDebugFlag, "Test message");
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/base/trace.hh:253:5: note: expanded from macro 'DPRINTF_UNCONDITIONAL'
    GEM5_DEPRECATED_MACRO_STMT(DPRINTF_UNCONDITIONAL,      \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/base/compiler.hh:93:24: note: expanded from macro 'GEM5_DEPRECATED_MACRO_STMT'
     do {{definition;} GEM5_DEPRECATED_MACRO(name, {}, message);} while (0)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/base/compiler.hh:89:76: note: expanded from macro 'GEM5_DEPRECATED_MACRO'
     ([](){[[deprecated(message)]] int name{}; return name;}(), (definition))
```

Reply:
Turns out it was a lot easier to fix than expected so I've done that now.

Reply:
Awesome, thanks!